projects/driver_controls/src/throttle.c

```diff
@@ -0,0 +1,153 @@
+// The module operates based on the voltag readings that come from the pedals.
+// The pedal occupies two channels of the ADS1015 one of which is chosen to be the "main" channel.
+// Everytime ADS1015 finishes a conversion of the pedal input on the main channel,
+// prv_flag_update_callback is called to set two flags accordingly which determine if
+// the readings are up tp date. It also verifies if the second channel is in synch with the main.
+// A periodic safety check, prv_raise_event_timer_callback, checks for
+// these flags and updates the position of the pedal in throttle_storage and raises the events
+// related to the pedal's position(braking, coasting, and accelerating zones).
+// If the data turns out to be stale, a timeout event will be raised.
+// The throttle_get_position function simply reads the position from throttle_storage.
+#include "throttle.h"
+#include <string.h>
+#include "event_queue.h"
+#include "input_event.h"
+
+// The time period between every update of the pedal readings.
+#define THROTTLE_UPDATE_PERIOD_MS 50
+
+// Scales a reading to a measure out of 12 bits based on the given range.
+#define THROTTLE_SCALE_READING_TO_12_BITS(reading, upper_bound, lower_bound) \
+  ((1 << 12) * ((reading) - (lower_bound)) / ((upper_bound) - (lower_bound)))
+
+// The tolerance for verifying channel readings match their supposed relationship.
+// This tolerance is defined for the scaled-to-12bits measures.
+#define THROTTLE_CHANNEL_SCALED_READINGS_TOLERANCE 10
```

Comment:
This is probably something that should have some math behind it - it'll depend on amount you're scaling each value since won't be anywhere near continuous functions.